examples/binary_mclp_distance_matrix_example.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: UTF-8 -*-
+
+import logging
+import sys
+import arcpy
+import pulp
+import csv
+import sys
+import os
+from pyspatialopt.analysis import arcpy_analysis
+from pyspatialopt.models import utilities
+from pyspatialopt.models import covering
+from pyspatialopt.models import binary_mclp_distance_matrix
+
+if __name__ == "__main__":
+    # Initialize a logger so we get formatted output
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    formatter = formatter = logging.Formatter('%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+    # setup stream handler to console output
+    sh = logging.StreamHandler(sys.stdout)
+    sh.setFormatter(formatter)
+    logger.addHandler(sh)
+
+    import os
+    scriptDir = os.path.dirname(os.path.realpath(__file__))
+
+    # a distance matrix file (format of csv)
+
+    workspace_path = r"../sample_data"
+    file_distance_matrix = r'service_area_demand_point_distance_matrix.csv'
+    service_dist = 5000
+    num_facility = 5
+
+    # This file should contain the following fields: facility_id, demand_id, demand, distance
+    list_field_req = ["facility_id", "demand_id", "demand", "distance"]
+
+    # do not change this variable unless you understand what it is
+    facility_variable_name = "facility"
+    # read the distance matrix
+    with open(os.path.join(workspace_path, file_distance_matrix)) as csvfile:
+        dict_pairwise_distance = [
+            {k: v for k, v in row.items()}
+            for row in csv.DictReader(csvfile, skipinitialspace=True)
+            ]
+
+    # test if it contains the required field
+    item_pairwise_distance = dict_pairwise_distance[1]
+
+    for field in list_field_req:
+        if field not in item_pairwise_distance.keys():
+            logger.info("Error: this field {} not found in the distance csv".format(field))
+            sys.exit(0)
```

Comment:
sys isn't imported in this file . Sorry, I may have mad you remove that earlier.